rasa/core/actions/forms.py

```diff
@@ -308,7 +308,10 @@ def _user_rejected_manually(self, validation_events: List[Event]) -> bool:
     def _get_events_since_last_user_uttered(
         tracker: "DialogueStateTracker",
     ) -> List[SlotSet]:
+        # Why would latest message not be in events?
         if tracker.latest_message in tracker.events:
+            # This will find first event with same text (since entities and intents
```

Comment:
Good spot with `UserUttered` equality! I did not foresee the problem of duplicate messages with same text. Probably this needs to be fixed also to find the `latest` one, not the first one. But check first if removing the problem with duplicate `SlotSet` events fixes this, it would be awesome if there would be a unit test for `/restart` too, this way the CI would have caught this issue early.